arangod/Cluster/ActionDescription.cpp

```diff
@@ -26,16 +26,17 @@
 #include "Basics/StaticStrings.h"
 
 #include <functional>
+#include <boost/functional/hash.hpp>
 
 using namespace arangodb;
 using namespace arangodb::maintenance;
 
 /// @brief ctor
-ActionDescription::ActionDescription(std::map<std::string, std::string> const& d,
+ActionDescription::ActionDescription(std::map<std::string, std::string>&& d,
```

Comment:
I dislike taking arguments by `&&` because this forces the caller to move something into your function. Receive it by value instead, then the caller can decide if she wants to copy or move. Is the speed improvement by eliminating one move constructor (which visible in this compile unit and thus mostly optimized out) worth it?

And by the way, is there a reason why this has to be an ordered map? O_o I think those action descriptions are only ever accessed by point lookup. I know it wasn't you, but maybe we can fix it now.

Reply:
I changed the constructor's signature so it wants the description map by-value.
I don't want to change from an ordered std::map to something unordered here. It seems you are right, and we only access individual keys from the description, so something unordered would work. But we are using the description also for creating a hash of the ActionDescription, and this hash is currently depending on the order of the description items. So I am a bit hesitant to change this, because then I would need to make the hash function order-agnostic, too.